### ISSUE TITLE
ProtectedBrowserStorage issue with key from DataProtection

### ISSUE THREAD

**Issue by johanskoldekrans** [aspnet/AspLabs#241]
Everytime I release a new pipeline we get this error.

I have specified a path in Startup for DataProtection to make sure the keys are intact but I still get this error. Ideas?

`System.Security.Cryptography.CryptographicException: The key {5cf41a4f-53bb-49af-9577-211e26dd54ad} was not found in the key ring.
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.UnprotectCore(Byte[] protectedData, Boolean allowOperationsOnRevokedKeys, UnprotectStatus& status)
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.DangerousUnprotect(Byte[] protectedData, Boolean ignoreRevocationErrors, Boolean& requiresMigration, Boolean& wasRevoked)
   at Microsoft.AspNetCore.DataProtection.KeyManagement.KeyRingBasedDataProtector.Unprotect(Byte[] protectedData)
   at Microsoft.AspNetCore.DataProtection.DataProtectionCommonExtensions.Unprotect(IDataProtector protector, String protectedData)
   at Microsoft.AspNetCore.ProtectedBrowserStorage.ProtectedBrowserStorage.GetAsync[T](String purpose, String key)`

**Comment by johanskoldekrans** [comment]
It works if I start the site in an InPrivate window. But if I release again that window breaks and Ctrl+F5 doesn't work.

**Comment by Wolftousen** [comment]
We just encountered this too and why just had to scrap using this library all together.  We could have set up an AWS DataProtection provider (https://aws.amazon.com/blogs/developer/aws-ssm-asp-net-core-data-protection-provider/), but we opted not too.

We are planning on going full client side Blazor later this year, so i'm not sure how viable this library will be in that regard.  The DataProtection library, by default, doesn't have persistence across deployments, so you'll get a new key every deploy unless you set up a provider.  

For client side blazor, there is no provider and you only have local storage anyway, so i'm not sure i see the point in using DataProtection when i'd have to store the encryption key stuff in the same place as the stuff i'm encrypting, not exactly secure.  Or i'd have to set up an API to provide the key constantly and then create a custom provider to retrieve the key, not sure it's worth it.
